dear_petition/petition/api/viewsets.py

```diff
@@ -44,14 +44,22 @@ class ContactViewSet(viewsets.ModelViewSet):
 
 
 class BatchViewSet(viewsets.ModelViewSet):
-
     queryset = petition.Batch.objects.prefetch_related(
         "petitions", "records__offenses__offense_records"
     )
     serializer_class = serializers.BatchSerializer
     permission_classes = [permissions.IsAuthenticated]
     parser_classes = (parsers.MultiPartParser, parsers.FormParser)
 
+    def get_queryset(self):
+        """ Filter queryset so that user's only have read access on objects they have created
+        """
+        if self.request.user.is_superuser:
```

Comment:
I might flip this, something like:
```suggestion
        qs = super().get_queryset()
        if not self.request.user.is_superuser:
            qs = qs.filter(user=...)
        return qs
```

Reply:
done